utils/mail.py

```diff
@@ -55,8 +55,8 @@ def wrapper(subject, email_body, email_from=None, email_to=list(), reply_to=None
             email_to = [email_to]
 
         from accounts.models import SameUser
-        emails_mapping = {su.secondary_trans_email: su.orig_email for su in SameUser.objects.all()}
-        email_to = list(set([emails_mapping.get(email, email) for email in email_to]))
+        user_mapping = {su.secondary_user: su.main_user for su in SameUser.objects.all()}
+        email_to = list(set([user_mapping.get(user, user) for user in email_to]))  # strings are going to be left as is
```

Comment:
We originally made this decorator because we needed a way to go from a user -> modified email address -> user -> real email address.
Now that we have a user object passed in as the `to` field, we could do this in a nicer way. We can use `user.profile.get_sameuser_object().orig_mail` (catch exception) to get the correct email to send to if the user is one of these merged email users.
If this is the case, then we can remove the decorator and put this logic inside `send_mail` where you check `email_is_valid`.

After discussing with @ffont we decided that it's not a great idea to make `email_to` accept either users or email addresses. We think that there is only one place where we have to use an email address (in the email change form). In this case, we could have a separate parameter (`raw_email_to`?) which contains these email addresses. Allow this function to only be called with email_to or raw_email_to.

Reply:
For the record there are several more places where the `send_email` function is used with the admin/support email addresses from `config.py`.

Reply:
I wonder if we could split this into 2 methods, send_email_to_user and send_email_to_admin (which either reads the admin email addresses from settings, or read email addresses from the user profile and then calls a generic send_email method.